TROUBLESHOOTING.md

```diff
@@ -0,0 +1,155 @@
+# Troubleshooting performance issues
+
+Performance is at the heart of Quarkus.
+
+If you are facing performance issues (runtime or startup issues), and would like to discuss them with the Quarkus Team,
+you are more than welcome on our [mailing list](https://groups.google.com/d/forum/quarkus-dev),
+[Zulip chat](https://quarkusio.zulipchat.com) or [Github issue tracker](https://github.com/quarkusio/quarkus/issues).
+
+To help us to troubleshoot your issues, we will need some performance insight from your application.
+
+On Linux or macOS, one of the best way to gather performance insight would be to generate CPU and allocation [FlameGraphs](https://github.com/brendangregg/FlameGraph) 
+via [Async Profiler](https://github.com/jvm-profiling-tools/async-profiler).
+
+If you want a deeper introduction to Async Profiler, go read [this article](https://hackernoon.com/profiling-java-applications-with-async-profiler-049s2790).
+
+## Installing Async Profiler
+
+To install Async Profiler, go to the [release page](https://github.com/jvm-profiling-tools/async-profiler/releases) and download the latest release.
+
+Async Profiler depends on `perf_events`. <br/>
+To allow capturing kernel call stacks using `perf_events` from a non-root process,
+you must make some OS configuration.
+
+For your session:
+
+```shell script
+echo 1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+```
+
+Or definitively using `sysctl`:
+
+```shell script
+sudo sysctl -w kernel.perf_event_paranoid=1
+sudo sysctl -w kernel.kptr_restrict=0
+```
+
+For allocation profiling, you also need to install HotStop debug symbol.
+
+Depending on your Linux and Java distribution this can be done via:
+
+```shell script
+# Ubuntu/Debian - Java 8
+apt install openjdk-8-dbg
+
+# Ubuntu/Debian - Java 11
+ apt install openjdk-11-dbg
+
+# On CentOS, RHEL and some other RPM-based distributions
+debuginfo-install java-1.8.0-openjdk
+```
+
+If needed, see this section in the documentation site for details: [allocation profiling](https://github.com/jvm-profiling-tools/async-profiler#allocation-profiling).
+
+## Profiling application runtime with Async Profiler
+
+Asych Profiler comes with a Java agent, and a command line.
+
+To profile application during its runtime, I recommend using the command line as you can choose when to start the profiler, and avoid your profile to be bloated with startup events. <br/>
+Indeed, when an application start, it does a lot a bootstraping instruction, that will no more occur during the rest of the run of your application. <br/>
+By starting the profiling when needed, you avoid these bootstrap instructions to be part of the profile.
+
+When you profile an application runtime, it's good to create some load on your application using a load generator tool (wrk, gatling, ...),
```

Comment:
Maybe we  should have some links for wrk(2), gatling, hyperfoil etc...

Reply:
There is so many tools and some are paid so I don' want us to promote some over others ...
Adding links seems to promote these tools.

Reply:
I don't agree, but I won't fight it :). 

For example we probably do want to be linking to wrk2 instead of wrk, it's too easy for people to get confused.

Reply:
OK, so maybe I can add some commonly tools with there links: ab, wrk2, Gatling, Apache JMeter at least ...

Reply:
Sounds good to me, :+1:

Reply:
Updated

Reply:
If are going to link to tools, we want to link to tools that we know do not suffer from co-ordination ommission problems, otherwise we end up with issues like this: https://github.com/quarkusio/quarkus/issues/9202

Reply:
Maye this should be a separate doc with list of tools that we use, and how to use them?

Reply:
I can remove `ab` from the list.

As I previously said, load testing is a vast and debatable subject.

There is multiple type of tools (URL basher, scenario based load testing, ...), some are paid some are free, some uses complex injectors, ...
And using these tools is far from easy ...

I added these ones because these are the one I'm used to, but any will works to generates a profile (not any will works to simulate correctly user interaction).

I'm open to suggestion here, we can add a section about them or create a dedicated page for them later.

We also have the performance page where we can add some section to tools used to generate load: https://quarkus.io/guides/performance-measure
Maybe we need to link this page inside the performance page ?